src/kmerseek/search.py

```diff
@@ -0,0 +1,286 @@
+
+    def read_results(self):
+        df = pl.scan_csv(self.output_csv)
+        return df
+
+    def filter_target_kmers_in_results(self):
+        # This might not actually be necessary since join_query_target_kmers
+        # would do the filter implicitly by only taking the inner matches
+        self.target_kmers_in_results = self.target.kmers_lazyframe.filter(
+            pl.col("sequence_name").is_in(self.results["match_name"])
+        )
+
+    def _prep_kmers_for_merging(self, kmers, suffix):
+        cols_to_rename = ["kmer", "start", "sequence_name", "sequence_file"]
+
+        renamer = {x: f"{x}{suffix}" for x in cols_to_rename}
+        kmers_renamed = kmers.rename(renamer).sort(self.join_query_target_kmers_on)
+        return kmers_renamed
+
+    def join_query_target_kmers(self):
+
+        query_kmers_prepped = self._prep_kmers_for_merging(
+            self.query.kmers_lazyframe, "_query"
+        )
+        target_kmers_prepped = self._prep_kmers_for_merging(
+            self.target.kmers_lazyframe, "_match"
+        )
+
+        # No collect yet
+        self.kmers = query_kmers_prepped.join(
+            target_kmers_prepped, on=self.join_query_target_kmers_on
+        )
+
+    def compute_tf_idf(self):
+        raise NotImplementedError()
+
+    def join_results_kmers(self):
+
+        self.results_with_kmers = self.results.join(
+            self.kmers,
+            left_on=self.join_results_search_on,
+            right_on=self.join_results_kmers_on,
+        )
+
+    def stitch_kmers_per_gene(self):
+        # Define the schema for the output of stitch_kmers_in_query_match_pair
+        output_schema = {
+            "match_name": pl.Utf8,
+            "query_name": pl.Utf8,  # Add schema for query_name
+            "query_start": pl.UInt32,
+            "query_end": pl.UInt32,
+            "query": pl.Utf8,
+            "match_start": pl.UInt32,
+            "match_end": pl.UInt32,
+            "match": pl.Utf8,
+            "to_print": pl.Utf8,
+        }
+
+        self.per_gene_stitched_kmers = (
+            self.results_with_kmers.group_by("match_name")
+            .map_groups(stitch_kmers_in_query_match_pair, schema=output_schema)
+            .sort(["query_start", "query_end"])
+        ).collect()
+
+    def show_results_per_gene(self):
+        cols = self.per_gene_stitched_kmers.columns
+        for row_data in self.per_gene_stitched_kmers.iter_rows():
+            # Each row is a single match
+            row = dict(zip(cols, row_data))
+            # import pdb
+
+            # pdb.set_trace()
+            print(f"\n---\nQuery Name: {row['query_name']}")
+            print(f"Match Name: {row['match_name']}")
+            print(row["to_print"])
+
+
+@click.command()
+@click.argument("query_fasta")
+@click.argument("target_fasta")
+@click.option("--moltype", default="hp")
+@click.option("--ksize", default=24)
+@click.option("--scaled", default=5)
+@click.option("--output", default="kmerseek_search.csv")
```

Comment:
Is output to `stdout` an option?

Reply:
`kmerseek search ... | cut -f 2,3 | grep -v foo | zstd --adapt -o filtered.txt.zst` is a kind of filtering later 😉